sql-plugin/src/main/scala/org/apache/spark/sql/catalyst/json/rapids/GpuJsonScan.scala

```diff
@@ -138,6 +140,11 @@ object GpuJsonScan {
       meta.willNotWorkOnGpu("GpuJsonScan only supports \"\\n\" as a line separator")
     }
 
+    parsedOptions.encoding.foreach(enc =>
+      if (enc != StandardCharsets.UTF_8.name() && enc != StandardCharsets.US_ASCII.name()) {
+      meta.willNotWorkOnGpu("GpuJsonScan only supports UTF8 or US-ASCII encoded data")
```

Comment:
Could we have the error message include the encoding that is causing us to fall back? This will make it easier for the user to debug.

Reply:
Thx for the comment, I'd like to create a follow-up targeting 22.04